sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -203,223 +156,218 @@ public BoundedJdbcSource withPassword(String password) {
     @Nullable
     private final String password;
 
-    public BoundedJdbcSource(DataSource dataSource, String query, String username,
-                             String password) {
+    private Read(DataSource dataSource, String query, @Nullable String username,
```

Comment:
Don't duplicate variables between the Read class and the JdbcOptions class. Instead make Read have a private variable of type JdbcOptions.